arch/rv32i/src/clint.rs

```diff
@@ -0,0 +1,66 @@
+// Note that it looks like this has been replaced MMIO-wise by machine_timer. Still useful for
+// hifive1 board. Perhaps this, machine_timer, and the clic.rs belong under boards?
+use kernel::common::registers::{ReadOnly, ReadWrite};
+use kernel::common::StaticRef;
+
+// memory map as described here: https://sifive.cdn.prismic.io/sifive%2F898b5153-4c06-4085-8d95-2d5fd67e74c6_u54_core_complex_manual_v19_02.pdf
+#[repr(C)]
+struct ClintRegisters {
+    // Interrupt Pending Register
+    misp: ReadWrite<u32, MISP::Register>,
+    _reserved0: [u32; (0x200_4000 - 0x200_0004) / 0x4],
+    mtimecmp1: ReadWrite<u32, MTIME::Register>,
+    mtimecmp2: ReadWrite<u32, MTIME::Register>,
+    _reserved1: [u32; (0x200_bff8 - 0x200_4008) / 0x4],
+    mtime1: ReadWrite<u32, MTIME::Register>,
+    mtime2: ReadWrite<u32, MTIME::Register>,
+}
+
+register_bitfields![u32,
+    MISP [
+        MISPBIT OFFSET(0) NUMBITS(1) []
+    ],
+    MTIME [
+        MTIMEBITS OFFSET(0) NUMBITS(32) []
+    ]
+];
+
+const CLINT_BASE: StaticRef<ClintRegisters> =
+unsafe { StaticRef::new(0x200_0000 as *const ClintRegisters) };
+
+pub unsafe fn read_mtimecmp() -> u64 {
```

Comment:
These functions don't _need_ to be marked unsafe. Is there a reason they _should_ be considered unsafe?